### PR TITLE
[TIR][Schedule] Support padding-by-factor in PadEinsum

### PR DESCRIPTION
**Pull request by junrushao**
This PR introduces a minor behavior change in Pad-Einsum that makes it support padding-by-factor instead of padding-by-constant-number.

### CODE REVIEW

File: src/tir/schedule/primitive.h [hunk 1]
```diff
@@ -632,8 +632,7 @@ TVM_DLL StmtSRef DecomposePadding(ScheduleState self, const StmtSRef& block_sref
  * \param block_sref The block sref that matches the Einsum pattern.
  * \param padding The padding for each block iter.
  */
-TVM_DLL void PadEinsum(ScheduleState self, const StmtSRef& block_sref,
-                       const Array<Integer>& padding);
+TVM_DLL void PadEinsum(ScheduleState self, StmtSRef block_sref, Array<Integer> padding);
```
Comment (vinx13) [inline src/tir/schedule/primitive.h:635]: use const ref?

**Review by vinx13 — COMMENTED** [review]
could you also update example here https://github.com/apache/tvm/blob/main/python/tvm/tir/schedule/schedule.py#L3515

File: src/tir/schedule/primitive.h (hunk 1 above)
Comment (junrushao) [reply src/tir/schedule/primitive.h:635]: Fixed! There's no fundamental difference though.

**Comment by junrushao** [comment]
Thanks @vinx13 for the huge help fixing the tensor core case!

**Comment by junrushao** [comment]
Fixed a merge conflict

**Review by spectrometerHBH — APPROVED** [review]

**Pull request merged**